t/mojolicious/exception_lite_app.t

```diff
@@ -126,149 +126,213 @@ get '/dead_helper';
+  $t->get_ok('/dead_action.xml')->status_is(500)->content_type_is('application/xml')->content_is("<very>bad</very>\n");
+};
+
+subtest 'Dead action with unsupported format' => sub {
+  $t->get_ok('/dead_action.json')->status_is(500)->content_type_is('text/html;charset=UTF-8')
+    ->content_like(qr!get &#39;/dead_action&#39;!)->content_like(qr/dead action!/);
+};
+
+subtest 'Dead action with custom exception rendering' => sub {
+  $t->get_ok('/dead_action' => {Accept => 'text/plain'})->status_is(500)->content_type_is('text/plain;charset=UTF-8')
+    ->content_like(qr/^dead action!\n/);
+};
+
+subtest 'Action dies twice' => sub {
+  $t->get_ok('/double_dead_action_☃')->status_is(500)->content_like(qr!get &#39;/double_dead_action_☃&#39;!)
+    ->content_like(qr/File.+lite_app\.t\", line \d/)->content_like(qr/double dead action!/);
+};
+
+subtest 'Trapped exception' => sub {
+  $t->get_ok('/trapped')->status_is(200)->content_is('bar');
+};
+
+subtest 'Another trapped exception' => sub {
+  $t->get_ok('/trapped/too')->status_is(200)->content_is('works');
+};
+
+subtest 'Custom exception handling' => sub {
+  $t->get_ok('/custom')->status_is(200)->content_is('Custom handling works!');
+};
+
+subtest 'Exception in helper' => sub {
+  $t->get_ok('/dead_helper')->status_is(500)->content_like(qr/dead helper!/);
+};
+
+subtest 'Missing template' => sub {
+  $t->get_ok('/missing_template')->status_is(404)->content_type_is('text/html;charset=UTF-8')
+    ->content_like(qr/Page not found/);
+};
+
+subtest 'Missing template with different format' => sub {
+  $t->get_ok('/missing_template.xml')->status_is(404)->content_type_is('application/xml')
+    ->content_is("<somewhat>bad</somewhat>\n");
+};
+
+subtest 'Missing template with unsupported format' => sub {
+  $t->get_ok('/missing_template.json')->status_is(404)->content_type_is('text/html;charset=UTF-8')
+    ->content_like(qr/Page not found/);
+};
+
+subtest 'Missing template with custom rendering' => sub {
+  $t->get_ok('/missing_template.txt')->status_is(404)->content_type_is('text/plain;charset=UTF-8')
+    ->content_is('Missing template, whatever.');
+};
+
+subtest 'Missing template (failed rendering)' => sub {
+  $t->get_ok('/missing_template/too')->status_is(404)->header_is('X-Not-Found' => 1)
+    ->content_type_is('text/html;charset=UTF-8')->content_like(qr/Page not found/);
+};
+
+subtest 'Missing helper (correct context)' => sub {
+  $t->get_ok('/missing_helper')->status_is(500)->content_type_is('text/html;charset=UTF-8')
+    ->content_like(qr/Server error/)->content_like(qr/shift-&gt;missing_helper/);
+};
+
+subtest 'Reuse exception' => sub {
+  ok !$exception, 'no exception';
+
+  ok !$snapshot, 'no snapshot';
+
+  $t->get_ok('/reuse/exception')->status_is(500)->content_like(qr/Reusable exception/);
+
+  isa_ok $exception, 'Mojo::Exception', 'right exception';
+
+  like $exception, qr/Reusable exception/, 'right message';
+
+  is $snapshot->{foo}, 'bar', 'right snapshot value';
+
```

Comment:
Those empty lines don't add any value.

Reply:
Was just following your exact direction:

>Block comments become the subtest description, and then we add a blank line between the different test cases within the subtest.

I can revert to no spaces, no problem.

Reply:
@kraih should I undo the extra newlines under `Bundled static files` subtest as well?

Reply:
Updated the sentence, since you've taken it a bit too literal. And yes, blank lines only if there's a pattern to follow, like 3 repetitive groups of similar tests (as is the case in the #1579 example).

Reply:
Thanks, I've adjusted the newlines based on context, let me know if that's better or if I should adjust more :muscle: